reana_server/rest/users.py

```diff
@@ -340,3 +351,95 @@ def _logout():
         return resp
     else:
         return jsonify(message='User not logged in'), 401
+
+
+@blueprint.route('/token', methods=['PUT'])
+def request_token():
+    r"""Endpoint to request user access token.
+
+    ---
+    put:
+      summary: Requests a new access token for the authenticated user.
+      description: >-
+        This resource allows the user to create an empty REANA access token
+        and mark it as requested.
+      operationId: request_token
+      produces:
+        - application/json
+      parameters:
+        - name: access_token
+          in: query
+          description: API access_token of user.
+          required: false
+          type: string
+      responses:
+        200:
+          description: >-
+            User information correspoding to the session cookie sent
+            in the request.
+          schema:
+            type: object
+            properties:
+              reana_token_status:
+                type: string
+          examples:
+            application/json:
+              {
+                "reana_token_status": "requested"
+              }
+        401:
+          description: >-
+            Error message indicating that the uses is not authenticated.
+          schema:
+            type: object
+            properties:
+              error:
+                type: string
+          examples:
+            application/json:
+              {
+                "error": "User not logged in"
+              }
+        403:
+          description: >-
+            Request failed. User token not valid.
+          examples:
+            application/json:
+              {
+                "message": "Token is not valid."
+              }
+        500:
+          description: >-
+            Request failed. Internal server error.
+          examples:
+            application/json:
+              {
+                "message": "Internal server error."
+              }
+    """
+    try:
+        user = None
+        if current_user.is_authenticated:
+            user = _get_user_from_invenio_user(current_user.email)
+        elif "access_token" in request.args:
+            user = get_user_from_token(request.args.get('access_token'))
+        user.request_access_token()
+        email_subject = f'[{REANA_URL}] Token request ({user.email})'
+        fields = ['id_', 'email', 'full_name', 'username', 'access_token',
+                  'access_token_status']
+        email_body = (
+            'New user access token request:\n\n' +
+            '\n'.join([f'{f}: {getattr(user, f, None)}' for f in fields]))
+        # TODO: change receiver
```

Comment:
How should we handle this?

Reply:
We could use [this variable](https://github.com/reanahub/reana/pull/309/files#diff-78c29e3abaa72a3c7f71bc5783c8c40bR38). We can rename it to `DEFAULT_EMAIL_RECEIVER` or `ADMIN_EMAIL`, what do you think?

Reply:
Done. We have to create the `ADMIN_EMAIL` env in the reana-server helm chart I guess.